### PR TITLE
Fixing some syntax errors to get actions working

### PR DESCRIPTION
**Pull request by carterbox**
Did some bug hunting in the actions to fix syntax errors. Please squash all of these useless commit messages when merging!

### CODE REVIEW

**Comment by carterbox** [comment]
P.S. If you want to test the changes in the PR before merging, you need to do it from my fork because actions are run from the what's in the main branch.

**Comment by BastianZim** [comment]
Hi @carterbox thanks for your contribution, looks great!! Unfortunately, I am currently travelling with very limited internet so I'm unable to really review this but I have added you as a collaborator so please feel free to merge this yourself and test it as well. 

I will be back in slightly more than a week but in case you finish everything before that, please feel also free to open a new PR in staged recipes and copy/paste your changes there. I'm just using this as a test ground and will close the repo once it's upstreamed.

No worries about commits as this is just for testing, I would recommend adding the features to staged-recipes with a nicer commit history in a separate PR anyways.

File: .github/workflows/add-author-issue_comment.yml [hunk 1]
```diff
@@ -7,27 +7,36 @@
...
+        github.event.issue.pull_request
+        && (github.event.sender.login != github.event.issue.user.login)
+        && contains(github.event.issue.labels.*.name, 'review-requested')
+        && !contains(github.event.issue.labels.*.name, 'Awaiting author contribution')
       runs-on: ubuntu-latest
       steps:
         - name: author-commented
           uses: actions/github-script@v6
           with:
+            debug: true
```
[the remark this answers is not in the document]
Comment (BastianZim) [reply .github/workflows/add-author-issue_comment.yml:20]: Thanks! I added the env variables but doesn't seem to be working.

File: .github/workflows/add-author-issue_comment.yml [hunk 2]
```diff
@@ -7,27 +7,36 @@
...
             script: |
+              // FIXME: Requesting team membership requires authentication/pemissions
+              var reviewer_is_staged_recipes_member = true;
               github.rest.teams.getMembershipForUserInOrg({
-                  org: 'conda-forge,
+                  org: 'conda-forge',
                   team_slug: 'staged-recipes',
-                  username: github.event.sender.login
+                  username: '${{ github.event.sender.login }}'
```
[the remark this answers is not in the document]
Comment (BastianZim) [reply .github/workflows/add-author-issue_comment.yml:27]: Yes could be. Quite some copy/pasting in the end on my side... 😄

File: .github/workflows/add-author-pr.yml [hunk 3]
```diff
@@ -11,27 +11,35 @@
...
-                  repo: context.repo.repo,
-                  labels: ['review-requested']
-              });
-              github.rest.issues.addLabels({
-                  issue_number: context.issue.number,
-                  owner: context.repo.owner,
-                  repo: context.repo.repo,
-                  labels: [label, 'Awaiting author contribution']
-              })
+              if (reviewer_is_staged_recipes_member) {
```
[the remark this answers is not in the document]
Comment (BastianZim) [reply .github/workflows/add-author-pr.yml:32]: Yes I hadn't gotten to that part, thanks for taking that on. 

Authentication can be added with a token: https://github.com/actions/github-script#using-a-separate-github-token

This needs to be from a user in conda-forge with the necessary rights to see team members. We'll need to check if we can use a conda-forge bot here or ask core if they can create one.

File: .github/workflows/add-review-team.yml [hunk 4]
```diff
@@ -7,15 +7,29 @@
...
+                '@conda-forge/help-go',
+                '@conda-forge/help-java',
+                '@conda-forge/help-julia',
+                '@conda-forge/help-nodejs',
+                '@conda-forge/help-perl',
+                '@conda-forge/help-python',
+                '@conda-forge/help-python-c',
+                '@conda-forge/help-r',
+                '@conda-forge/help-ruby'
+              ];
```
Comment (BastianZim) [inline .github/workflows/add-review-team.yml:32]: Slightly related: We need to ask core to create the missing teams and then see if somebody volunteers for them. Just as a mental TODO.

File: .github/workflows/add-author-issue_comment.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (BastianZim) [reply .github/workflows/add-author-issue_comment.yml:20]: Yes this is awesome, thanks! Somehow, GitHub Actions wasn't evaluating my `if` properly but that should help us a lot!

**Pull request merged**